x-pack/plugins/security_solution/public/management/components/endpoint_response_actions_list/components/actions_log_table.tsx

```diff
@@ -0,0 +1,378 @@
+
+import type { CriteriaWithPagination } from '@elastic/eui';
+import {
+  EuiI18nNumber,
+  EuiAvatar,
+  EuiBasicTable,
+  EuiButtonIcon,
+  EuiFacetButton,
+  EuiHorizontalRule,
+  RIGHT_ALIGNMENT,
+  EuiScreenReaderOnly,
+  EuiText,
+  EuiToolTip,
+  type HorizontalAlignment,
+} from '@elastic/eui';
+import { euiStyled } from '@kbn/kibana-react-plugin/common';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import type { ActionListApiResponse } from '../../../../../common/endpoint/types';
+import type { EndpointActionListRequestQuery } from '../../../../../common/endpoint/schema/actions';
+import { FormattedDate } from '../../../../common/components/formatted_date';
+import { TABLE_COLUMN_NAMES, UX_MESSAGES } from '../translations';
+import { getActionStatus, getUiCommand } from './hooks';
+import { getEmptyValue } from '../../../../common/components/empty_value';
+import { StatusBadge } from './status_badge';
+import { ActionsLogExpandedTray } from './action_log_expanded_tray';
+import { useTestIdGenerator } from '../../../hooks/use_test_id_generator';
+import { MANAGEMENT_PAGE_SIZE_OPTIONS } from '../../../common/constants';
+import { useActionHistoryUrlParams } from './use_action_history_url_params';
+import { useUrlPagination } from '../../../hooks/use_url_pagination';
+
+const emptyValue = getEmptyValue();
+
+// Truncated usernames
+const StyledFacetButton = euiStyled(EuiFacetButton)`
+  .euiText {
+    margin-top: 0.38rem;
+    overflow-y: visible !important;
+  }
+`;
+
+interface ActionsLogTableProps {
+  error?: string;
+  'data-test-subj'?: string;
+  items: ActionListApiResponse['data'];
+  isFlyout: boolean;
+  loading: boolean;
+  onChange: ({
+    page: _page,
+  }: CriteriaWithPagination<ActionListApiResponse['data'][number]>) => void;
+  // pageSize: number;
+  queryParams: EndpointActionListRequestQuery;
+  showHostNames: boolean;
+  totalItemCount: number;
+}
+interface ExpandedRowMapType {
+  [k: ActionListApiResponse['data'][number]['id']]: React.ReactNode;
+}
+
+export const ActionsLogTable = memo<ActionsLogTableProps>(
+  ({
+    'data-test-subj': dataTestSubj,
+    error,
+    items,
+    isFlyout,
+    loading,
+    onChange,
+    queryParams,
+    showHostNames,
+    totalItemCount,
+  }) => {
+    const getTestId = useTestIdGenerator('response-actions-list');
+
+    const { pagination: paginationFromUrlParams } = useUrlPagination();
+    const { withOutputs, setUrlWithOutputs } = useActionHistoryUrlParams();
+
+    const [itemIdToExpandedRowMap, setItemIdToExpandedRowMap] = useState<ExpandedRowMapType>(
+      withOutputs
+        ? withOutputs.reduce<ExpandedRowMapType>((idToRowMap, actionId) => {
+            if (actionId.length) {
+              idToRowMap[actionId] = (
+                <ActionsLogExpandedTray action={items.filter((item) => item.id === actionId)[0]} />
+              );
+            }
+            return idToRowMap;
+          }, {})
+        : {}
+    );
```

Comment:
Keep in mind that this will only be set once - during the initialization of the `setState()` call. Subsequent re-renders will not execute this.

Reply:
That is right. This is to show details on open trays with page reload with URL params set